pkg/sendconfig/common_workflows.go

```diff
@@ -35,18 +37,30 @@ func UpdateKongAdminSimple(ctx context.Context,
 	deprecatedLogger logrus.FieldLogger,
 	kongConfig Kong,
 	enableReverseSync bool,
+	diagnostic util.ConfigDumpDiagnostic,
 ) ([]byte, error) {
 	// build the kongstate object from the Kubernetes objects in the storer
 	storer := store.New(*cache, ingressClassName, false, false, false, deprecatedLogger)
 	kongstate, err := parser.Build(deprecatedLogger, storer)
 	if err != nil {
 		return nil, err
 	}
+	var diagConfig *file.Content
 
 	// generate the deck configuration to be applied to the admin API
 	targetConfig := deckgen.ToDeckContent(ctx,
 		deprecatedLogger, kongstate,
 		kongConfig.PluginSchemaStore, kongConfig.FilterTags)
+	if diagnostic != (util.ConfigDumpDiagnostic{}) {
```

Comment:
```suggestion
	if diagnostic.(type) != (util.ConfigDumpDiagnostic{}) {
```

Reply:
This is checking to see if the argument we passed in is empty (hence the `{}`), not its type. It's empty when `--dump-config` isn't set. Adding a comment to clarify the check.